common/djangoapps/course_modes/models.py

```diff
@@ -367,6 +367,40 @@ def min_course_price_for_currency(cls, course_id, currency):
         modes = cls.modes_for_course(course_id)
         return min(mode.min_price for mode in modes if mode.currency == currency)
 
+    @classmethod
+    def has_professional_mode(cls, modes_dict):
+        """
+        check the course mode is profession or no-id-professional
+
+        Args:
+            modes_dict (dict): course modes.
+
+        Returns:
+            bool
+        """
+        return "professional" in modes_dict
+
+    @classmethod
+    def has_no_id_professional_mode(cls, modes_dict):
+        """
+        check the course mode is no-id-professional
+
+        Args:
+            modes_dict (dict): course modes.
+
+        Returns:
+            bool
+        """
+        return CourseMode.NO_ID_PROFESSIONAL_MODES[0] in modes_dict
+
+    @classmethod
+    def is_professional_mode(cls, course_mode_tuple):
+        return course_mode_tuple.slug == 'professional'
```

Comment:
I don't think you need separate methods for no-ID professional.  Outside of the `course_modes` app, we basically care about two things:

1) Is the course verified?  (True for "verified" and "professional")
2) Should the course be displayed as professional?  (True for "professional" and "no-id-professional")

We already have a method for (1), so I think you just need to add a classmethod for (2).